mesonbuild/compilers/mixins/clike.py

```diff
@@ -100,6 +100,21 @@ def to_native(self, copy: bool = False) -> T.List[str]:
                 new.insert(group_start, '-Wl,--start-group')
         # Remove system/default include paths added with -isystem
         default_dirs = self.compiler.get_default_include_dirs()
+        sys_root = None
+
+        for i, each in enumerate(new):
+            if each.startswith('--sysroot'):
+                if each == '--sysroot' and i + 1 < len(new):
+                    sys_root = new[i + 1]
+                    break
```

Comment:
I'd prefer to optimize for the common case, which is that `--sysroot` has a proper argument passed to it:
```python
if each == '--sysroot':
    try:
        sys_root = new[i + 1]
    except IndexError:
        mlog.warn('--sysroot passed to', self.language, 'compiler without an argument')
    break
```
Plus this will warn someone that they're probably doing something they didn't mean to.

Otherwise, this looks good to me.

Reply:
Sorry, forgot to actually push this after incorporating your suggestions and then forgot about it. I've rebased this and pushed out the updated commit.